### ISSUE TITLE
[droid] Nearest LockState calculated incorrectly

### ISSUE THREAD

**Issue by alexd-zyngr** [galadril/Xam.Plugin.SimpleBottomDrawer#11]
# GIVEN
Installed the nuget package of `SimpleBottomDrawer`

```c#
 public double[] LockStatesForSlidingBottomDrawer => _lockStatesForSlidingBottomDrawer;

        private readonly double[] _lockStatesForSlidingBottomDrawer =
            new double[]
            {
                  0 // Nothing visible
                , MapTabRootViewModel.DefaultVisiblePartOfStationInfo
                , 0.7
                , 0.9 //Entire Screen Under the Slider
                , 1
            };
```

```c#
 public double ExpandedPercentage
        {
            get => _expandedPercentage;
            set
            {
                _ = SetProperty(ref _expandedPercentage, value);

                if (value == DefaultVisiblePartOfStationInfo) //0.5625
                {
                    this.Logger.Debug("[ui] [bottom drawer] ExpandedPercentage !!! ACHTUNG !!!");
                }
                this.Logger.Debug("[ui] [bottom drawer] ExpandedPercentage={0}", value);

            }
        }
        private double _expandedPercentage = 0;
```

```xaml
<slidingdrawer:BottomDrawer
            x:Name="BottomDrawerView"
            LockStates="{
                Binding   Path=LockStatesForSlidingBottomDrawer 
                        , Source={x:Reference MapTabRootViewPage}
            }"
            IsExpanded="{
                Binding IsStationInfoPopoverVisible
                        , Mode=OneWay
            }"
            IsVisible="{
                Binding IsStationInfoPopoverVisible
                        , Mode=OneWay
            }"
            ExpandedPercentage="{
                Binding ExpandedPercentage
                        , Mode=TwoWay
            }"
            CornerRadius="16"
            RelativeLayout.HeightConstraint="{
                ConstraintExpression Type=RelativeToParent
                    , Property=Height
                    , Factor=1
                    , Constant=0
            }"
            RelativeLayout.WidthConstraint="{
                ConstraintExpression Type=RelativeToParent
                    , Property=Width
                    , Factor=1
                    , Constant=0
            }"
            RelativeLayout.YConstraint="{
                ConstraintExpression Type=RelativeToParent
                    , Property=Height
                    , Factor=1
                    , Constant=-40
            }"
            Padding="0">
```

# WHEN
1. Present `BottomDrawer` at `ExpandedPercentage==0.5625`
2. Slide up

# Actual result:
```
[Debug] [ui] [bottom drawer] ExpandedPercentage=0.553093169654548
[Debug] [ui] [bottom drawer] ExpandedPercentage=0.606566828722394
[Debug] [ui] [bottom drawer] ExpandedPercentage=0.628992091344056
[Debug] [ui] [bottom drawer] ExpandedPercentage=0.672685505957577
[Debug] [ui] [bottom drawer] ExpandedPercentage=0.714204585752008
[Debug] [ui] [bottom drawer] ExpandedPercentage=0.753656781585523
[Debug] [ui] [bottom drawer] ExpandedPercentage=0.772535185574153
[Debug] [ui] [bottom drawer] ExpandedPercentage=0.808014001260257
[Debug] [ui] [bottom drawer] ExpandedPercentage=0.841009129359069
[Debug] [ui] [bottom drawer] ExpandedPercentage=0.871174135687631
[Debug] [ui] [bottom drawer] ExpandedPercentage=0.897830643467397
[Debug] [ui] [bottom drawer] ExpandedPercentage=0.919919573394946
[Debug] [ui] [bottom drawer] ExpandedPercentage=0.939377001543951
[Debug] [ui] [bottom drawer] ExpandedPercentage=0.955736895513268
[Debug] [ui] [bottom drawer] ExpandedPercentage=0.968493498903413
[Debug] [ui] [bottom drawer] ExpandedPercentage=0.97352898453867
[Debug] [ui] [bottom drawer] ExpandedPercentage=0.981404821299974
[Debug] [ui] [bottom drawer] ExpandedPercentage=0.98546007891607
[Debug] [ui] [bottom drawer] ExpandedPercentage=0.990223463687151

>> [Debug] [ui] [bottom drawer] ExpandedPercentage=0.7
```

# Expected result:
```
[Debug] [ui] [bottom drawer] ExpandedPercentage=0.553093169654548
[Debug] [ui] [bottom drawer] ExpandedPercentage=0.606566828722394
[Debug] [ui] [bottom drawer] ExpandedPercentage=0.628992091344056
[Debug] [ui] [bottom drawer] ExpandedPercentage=0.672685505957577
[Debug] [ui] [bottom drawer] ExpandedPercentage=0.714204585752008
[Debug] [ui] [bottom drawer] ExpandedPercentage=0.753656781585523
[Debug] [ui] [bottom drawer] ExpandedPercentage=0.772535185574153
[Debug] [ui] [bottom drawer] ExpandedPercentage=0.808014001260257
[Debug] [ui] [bottom drawer] ExpandedPercentage=0.841009129359069
[Debug] [ui] [bottom drawer] ExpandedPercentage=0.871174135687631
[Debug] [ui] [bottom drawer] ExpandedPercentage=0.897830643467397
[Debug] [ui] [bottom drawer] ExpandedPercentage=0.919919573394946
[Debug] [ui] [bottom drawer] ExpandedPercentage=0.939377001543951
[Debug] [ui] [bottom drawer] ExpandedPercentage=0.955736895513268
[Debug] [ui] [bottom drawer] ExpandedPercentage=0.968493498903413
[Debug] [ui] [bottom drawer] ExpandedPercentage=0.97352898453867
[Debug] [ui] [bottom drawer] ExpandedPercentage=0.981404821299974
[Debug] [ui] [bottom drawer] ExpandedPercentage=0.98546007891607
[Debug] [ui] [bottom drawer] ExpandedPercentage=0.990223463687151

>> [Debug] [ui] [bottom drawer] ExpandedPercentage=1
(or at least 0.9 ; but not 0.7 definitely)
```

**Comment by alexd-zyngr** [comment]
P.S. I am sure my ViewModel logic does not overwrite that `ExpandedPercentage` binding since otherwise there would be some logs indicating that fact. So all writes in the description above are coming from within the nuget package.

**Comment by alexd-zyngr** [comment]
[UPD] seems like the library is capable of handling no more than 3 (three) values in a `LockStates` array. Continue investigating...

**Comment by dodikk** [comment]
Fixed in #12

**Comment by alexd-zyngr** [comment]
The mentioned PR has been merged, so this issue can be closed.
